src/scheduler.rs

```diff
@@ -30,6 +31,63 @@ pub fn run_scheduler(
     tokio::spawn(async move { scheduler_loop(config_store, checker, producer, shutdown).await })
 }
 
+fn record_result_metrics(result: &CheckResult) {
+    // Record metrics
+    let CheckResult {
+        status,
+        scheduled_check_time,
+        actual_check_time,
+        duration,
+        status_reason,
+        ..
+    } = result;
+
+    let status_label = match status {
+        CheckStatus::Success => "success",
+        CheckStatus::Failure => "failure",
+        CheckStatus::MissedWindow => "missed_window",
+    };
+    let failure_reason = match status_reason.as_ref().map(|r| r.status_type) {
+        Some(CheckStatusReasonType::Failure) => Some("failure"),
+        Some(CheckStatusReasonType::DnsError) => Some("dns_error"),
+        Some(CheckStatusReasonType::Timeout) => Some("timeout"),
+        None => None,
+    };
+
+    // Record duration of check
+    if let Some(duration) = duration {
+        metrics::histogram!(
+            "check_result.duration_ms",
+            "histogram" => "timer",
+            "status" => status_label,
+            "failure_reason" => failure_reason.unwrap_or("ok"),
+        )
+        .record(duration.num_milliseconds() as f64);
+    }
+
+    // Record time between scheduled and actual check
+    metrics::histogram!(
+        "check_result.delay_ms",
+        "histogram" => "timer",
+        "status" => status_label,
+        "failure_reason" => failure_reason.unwrap_or("ok"),
+    )
+    .record((*actual_check_time - *scheduled_check_time).num_milliseconds() as f64);
+
+    // Record status of the check
+    metrics::counter!(
+        "check_result.status",
```

Comment:
I think maybe this should be `check_result.processed` or something? The status is already a tag